src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java

```diff
@@ -731,4 +738,50 @@ private int getFirstLadder(@NotNull final BlockPos pos)
             return pos.getY() - 1;
         }
     }
+
+    /**
+     * Calculates the working position.
+     * <p>
+     * Takes a min distance from width and length.
+     * <p>
+     * Then finds the floor level at that distance and then check if it does contain two air levels.
+     * @param targetPosition the position to work at.
+     * @return BlockPos position to work from.
+     */
+    @Override
+    public BlockPos getWorkingPosition(BlockPos targetPosition)
+    {
+        return getNodeMiningPosition(targetPosition);
+    }
+
+    /**
+     * Create a save mining position for the miner.
+     *
+     * @param blockToMine block which should be mined or placed.
+     * @return the save position.
+     */
+    private BlockPos getNodeMiningPosition(BlockPos blockToMine)
+    {
+        if (getOwnBuilding() instanceof BuildingMiner)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless "instanceof" operator; it will always return "true". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1850)